devito/passes/iet/dtypes.py

```diff
@@ -1,14 +1,54 @@
 import numpy as np
 import ctypes
 
-from devito.ir import FindSymbols, Uxreplace
+from devito.ir import FindSymbols
+from devito.ir.iet.nodes import Dereference
+from devito.ir.iet.visitors import Uxreplace
+from devito.symbolics.extended_dtypes import Float16P
+from devito.tools.utils import as_list
+from devito.types.basic import Symbol
 
-__all__ = ['lower_complex']
+__all__ = ['lower_dtypes']
 
 
-def lower_complex(iet, lang, compiler):
+def lower_dtypes(iet, lang, compiler, sregistry):
     """
-    Add headers for complex arithmetic
+    Lowers float16 scalar types to pointers since we can't directly pass their
+    value. Also includes headers for complex arithmetic if needed.
+    """
+
+    iet, metadata = _complex_includes(iet, lang, compiler)
+
+    # Lower float16 parameters to pointers and dereference
+    body_prefix = []
+    body_mapper = {}
+    params_mapper = {}
+
+    # Lower scalar float16s to pointers and dereference them
+    for s in FindSymbols('scalars').visit(iet):
```

Comment:
looks like not, since the actual types I'm mapping (e.g. `Constant`) aren't nodes they are only caught by `FindSymbols` if it's as a reference within some other expression, so we need to visit the body. That said I can probably make this marginally more efficient by making a set of parameters beforehand and checking membership that way